libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionLoader.java

```diff
@@ -38,18 +30,14 @@
    * @param textView to populate with instruction
    * @param bannerText containing components to populate into textView
    */
-  public InstructionLoader(TextView textView, BannerText bannerText) {
-    this(textView, bannerText, ImageCoordinator.getInstance(), new AbbreviationCoordinator());
+  InstructionLoader(TextView textView, @NonNull BannerText bannerText) {
+    this(textView, new BannerComponentTree(bannerText, new ExitSignCreator(),
+      ImageCreator.getInstance(), new AbbreviationCreator(), new TextCreator()));
```

Comment:
I wonder if this order is the source of the crash seen in Firebase. 

If the order these `Creator`s are added affects the sequence of processing, this would be a possible issue because the images are loaded asynchronously.  

Is it possible for the `InstructionLoader` to move on to abbreviations / text before the images are loaded?  Causing the indices for the nodes to get out of order?  

For reference:

```
java.lang.IndexOutOfBoundsException: setSpan (21 ... 27) ends beyond length 23
FATAL EXCEPTION: main
Process: com.mapbox.services.android.navigation.app, PID: 9078
java.lang.IndexOutOfBoundsException: setSpan (21 ... 27) ends beyond length 23
	at android.text.SpannableStringInternal.checkRange(SpannableStringInternal.java:351)
	at android.text.SpannableStringInternal.setSpan(SpannableStringInternal.java:79)
	at android.text.SpannableString.setSpan(SpannableString.java:46)
	at com.mapbox.services.android.navigation.ui.v5.instruction.TextViewUtils.setImageSpan(TextViewUtils.java:52)
	at com.mapbox.services.android.navigation.ui.v5.instruction.TextViewUtils.setImageSpan(TextViewUtils.java:41)
	at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionTarget.createAndSetImageSpan(InstructionTarget.java:65)
	at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionTarget.onBitmapLoaded(InstructionTarget.java:38)
	at com.squareup.picasso.TargetAction.complete(TargetAction.java:36)
	at com.squareup.picasso.Picasso.deliverAction(Picasso.java:576)
	at com.squareup.picasso.Picasso.complete(Picasso.java:528)
	at com.squareup.picasso.Picasso$1.handleMessage(Picasso.java:122)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5221)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
```

Reply:
@danesfeder that's what I originally thought and I had re-arranged them but it did not fix the bug. I'm working on that bug now